clients/src/main/java/org/apache/kafka/common/record/RecordsProcessingStats.java

```diff
@@ -18,15 +18,15 @@
 
 public class RecordsProcessingStats {
 
-    public static final RecordsProcessingStats EMPTY = new RecordsProcessingStats(0L, 0L, -1);
+    public static final RecordsProcessingStats EMPTY = new RecordsProcessingStats(0L, 0, -1);
 
     private final long temporaryMemoryBytes;
-    private final long conversionCount;
+    private final int numRecordsConverted;
```

Comment:
Shouldn't this be of long type ?

Reply:
I don't think so. What would be the reason?

Reply:
The for loop structure involving line 112 of AbstractRecords.java doesn't change.
Increment by 1 is changed to increment by builder.numRecords().
Meaning the final outcome would be bigger than prior to change.

Hence I think this should keep the long type.

Reply:
The point was that `long` was unnecessary in the first place. We can never read more than `2 GB` from the log at a time:

```scala
def read(startOffset: Long, maxLength: Int, maxOffset: Option[Long] = None, minOneMessage: Boolean = false,
           isolationLevel: IsolationLevel): FetchDataInfo = {
```

Reply:
Good to know.